zebra/zebra_vxlan.c

```diff
@@ -4226,61 +4226,174 @@ void zebra_vxlan_remote_vtep_del(ZAPI_HANDLER_ARGS)
 				   &vtep_ip, vni,
 				   zebra_route_string(client->proto));
 
-		/* Locate VNI hash entry - expected to exist. */
```

Comment:
This check didn't get removed on the `zebra_vxlan_remote_vtep_ADD()` side. Should it of been? I think so, probably don't need to check this until it hits the actual code.

Reply:
I think this may be just an artifact of the display of the diff? both the "_add" and "_del" functions are split into two in the PR: a "_zapi" part that decodes the zapi and enqueues the update, and the actual "add" and "del" logic, that's called when the enqueued update is actually processed.

Reply:
sorry I think I explained poorly.

If you look in the vtep_add_zapi() it does this zevpn lookup and failure. But it does nothing with that variable after other than continue. Further, it does the exact same lookup in vtep_add() and vtep_del() when it's time to actually handle it and it actually uses the zevpn value then.

My feeling this check should have been removed and indeed vtep_del_zapi() doesn't have it. Since the value is never used and it's probably logic that should be saved for the handler anyway.

Reply:
Same is probably true for all the "basic" checks in the zapi code tbh. I feel any "checking" logic should be saved for when we hit the handler, this function should queue it up.

Reply:
Ok, thanks, I see what you're saying: yes, I'll move/remove those 'lookup' tests. There is precedent for some kinds of top-level, global tests though: some of the mpls/lsp zapi handlers check whether mpls is enabled, for example. I would like to leave some of those front-end tests in place to avoid the work of constructing and processing an update that is just ... not valid?

Reply:
hmm.. i guess the top level ones make sense to avoid queuing. At least for just checking if it's enabled.

Can you make the logs different then? There is an unlikely chance it could pass one and fail the other and right now they have the same log.

Reply:
ah, yes, good catch: yes, I'll make the logs distinguishable...